File: 0371_sum_of_two_integers/sum_of_two_ints.py
def get_sum(a: int, b: int) -> int:
    """
    Time: O(n)
    Space: O(1)   
    """
    # 32 bit integer max.
    MAX = 0x7FFFFFFF
    # Mask to get last 32 bits.
    mask = 0xFFFFFFFF
    while b:
        carry = (a & b)
        a = (a ^ b) & mask
        b = (carry << 1) & mask
    # (~) Converts big number into negative in Python.
    return a if a <= MAX else ~(a ^ mask)

File: 0371_sum_of_two_integers/sum_of_two_ints.cpp
#include <iostream>
#include <vector>

using namespace std;

/**
 * Time: O(n)
 * Space: O(n) 
 */
int get_sum(int a, int b)
{
    return b == 0 ? a : getSum(a ^ b, (unsigned int)(a & b) << 1);
}